### DISCUSSION TITLE
PI controlled valve setup

### DISCUSSION THREAD

**Discussion by ThomasCr** [jmcollin78/versatile_thermostat#275] [Q&A]
Hi,

I don't understand the setup for PI controlled valve, because I get a lot spikes:

![grafik](https://github.com/jmcollin78/versatile_thermostat/assets/226670/44bbeff6-d809-4a2e-90d8-724e55b5339c)

We can see, that some values are only present for a second and get changed really fast.

My setup is:

![grafik](https://github.com/jmcollin78/versatile_thermostat/assets/226670/6b2a2de8-f0a2-4c7f-a27c-288bef3b9e40)
![grafik](https://github.com/jmcollin78/versatile_thermostat/assets/226670/19039d50-1862-4125-9386-6c2a44ad77d7)
![grafik](https://github.com/jmcollin78/versatile_thermostat/assets/226670/4f3ae655-53f3-44b8-965b-5094f06cb0fb)
![grafik](https://github.com/jmcollin78/versatile_thermostat/assets/226670/69b88a5d-3fde-469d-a436-64b91e99556e)
![grafik](https://github.com/jmcollin78/versatile_thermostat/assets/226670/87ee54ea-0efa-4c97-986b-104da16a57a1)
![grafik](https://github.com/jmcollin78/versatile_thermostat/assets/226670/f0091fbf-fdd8-48d0-b84f-e8e3acd5c55f)

Have I set up anything wrong?

**Comment by jmcollin78** [comment]
Hello, @ThomasCr ,

Sadly I don't have the scale of the graphs so it is difficult to see if it is big spikes or only "graphical effects" but if I suppose that the first graph is between 0 and 100% it seems not normal. The other important think missing is the target temperature. If the target temperature change it is normal to have some spikes at the time of the change.

The most important think to debug is to have this kind of graph: https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#even-better-with-plotly-to-tune-your-thermostat

You will then have graphs like this which is easier to understand what happens :
![Capture d’écran 2023-12-17 à 15 52 43](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/250c7e90-ac8f-4d41-9df7-d8b93bd09f1b)

And give the me le legend and the axes please.
